src/state/selectors.js

```diff
@@ -105,24 +109,18 @@ export const getEmojiCounts = createSelector(
   }
 );
 
-export const getSelectedEmoji = createSelector(
-  getEmojiQuestion,
-  getSelected,
-  (question, selected) => question && question.options.find(option => option.id === selected.emoji)
-);
-
-export const getSelectedTopic = createSelector(
-  getMultipleChoiceQuestions,
-  getSelected,
-  (mcQuestions, selected) => {
-    const questionIds = Object.keys(mcQuestions);
-    for (let i = 0; i < questionIds.length; i += 1) {
-      const question = mcQuestions[questionIds[i]];
-      const match = question && question.options.find(option => option.id === selected.topic);
-      if (match) {
-        return match;
-      }
-    }
-    return null;
-  }
+/**
+ * Return an array of topic multiple-choice question objects ("topic" is the
+ * grouping question that isn't emoji) with counts for how often each emoji
+ * occurs in the response data
+ *
+ * @param {Object} state The state object
+ * @returns {Object[]} An array of `{ value, id, count }` objects
+ */
+export const getTopicCounts = createSelector(
+  getTopicList,
+  getAggregations,
+  (topicList, aggregations) => topicList.map(option => Object.assign({
+    count: aggregations[option.id].count
+  }, option))
```

Comment:
This one-liner (with formatting) replaces the nasty 25-line reducer that was `getMultipleChoiceCounts`.